standardization/src/main/scala/za/co/absa/enceladus/standardization/interpreter/stages/TypeParser.scala

```diff
@@ -92,29 +93,29 @@ object TypeParser extends StandardizationCommon {
                     path: String,
                     origSchema: StructType,
                     parent: Option[Parent] = None): TypeParser = {
-    val parserClass: (StructField, String, StructType, Option[Parent]) => TypeParser = field.dataType match {
-      case _: ArrayType => ArrayParser
-      case _: StructType => StructParser
-      case _: ByteType => IntegralParser(_, _, _, _,Set(ShortType, IntegerType, LongType), Byte.MinValue, Byte.MaxValue)
-      case _: ShortType => IntegralParser(_, _, _, _, Set(IntegerType, LongType), Short.MinValue, Short.MaxValue)
-      case _: IntegerType => IntegralParser(_, _, _, _, Set(LongType), Int.MinValue, Int.MaxValue)
-      case _: LongType => IntegralParser(_, _, _, _, Set.empty, Long.MinValue, Long.MaxValue)
-      case _: FloatType | _: DoubleType => FractionalParser
-      case _: DecimalType => DecimalParser
-      case _: StringType => StringParser
-      case _: BooleanType => BooleanParser
-      case _: DateType => DateParser
-      case _: TimestampType => TimestampParser
+    val parserClass: (String, StructType, Option[Parent]) => TypeParser = field.dataType match {
+      case _: ArrayType => ArrayParser(TypedStructField.asArrayTypeStructField(field), _, _, _)
+      case _: StructType => StructParser(TypedStructField.asStructTypeStructField(field), _, _, _)
+      case _: ByteType => IntegralParser(TypedStructField(field), _, _, _,Set(ShortType, IntegerType, LongType), Byte.MinValue, Byte.MaxValue)
+      case _: ShortType => IntegralParser(TypedStructField(field), _, _, _, Set(IntegerType, LongType), Short.MinValue, Short.MaxValue)
+      case _: IntegerType => IntegralParser(TypedStructField(field), _, _, _, Set(LongType), Int.MinValue, Int.MaxValue)
+      case _: LongType => IntegralParser(TypedStructField(field), _, _, _, Set.empty, Long.MinValue, Long.MaxValue)
+      case _: FloatType | _: DoubleType => FractionalParser(TypedStructField.asFractionalTypeStructField(field), _, _, _)
+      case _: DecimalType => DecimalParser(TypedStructField(field), _, _, _)
+      case _: StringType => StringParser(TypedStructField(field), _, _, _)
+      case _: BooleanType => BooleanParser(TypedStructField(field), _, _, _)
+      case _: DateType => DateParser(TypedStructField.asDateTimeTypeStructField(field), _, _, _)
```

Comment:
Did we already have a chat if these should be on the new line or not?
Like `...Type =>\n Array...`

Reply:
Honestly I don't remember if we did.
Tried, when multiline  the cases drag, decreasing the clarity. But I aligned the => arrows the way how you like 😉